torch/_library/utils.py

```diff
@@ -49,3 +51,43 @@ def parse_namespace(qualname: str) -> Tuple[str, str]:
             f"of a namespace and a name, e.g. aten::sin"
         )
     return splits[0], splits[1]
+
+
+def lookup_op(qualname: str) -> torch._ops.OpOverloadPacket:
+    namespace, name = parse_namespace(qualname)
+    if "." in name:
+        name, overload = name.split(".")
+    else:
+        overload = "default"
+    ns = getattr(torch.ops, namespace)
+    packet = getattr(ns, name)
+    return getattr(packet, overload)
+
+
+def is_functional_schema(schema: Any) -> bool:
+    """Check if the schema is functional.
+
+    An operator is functional if:
+    - it does not mutate any of its inputs
+    - it does not return a view on any of its inputs
+    - it has at least one return
+    """
+
+    # Lazy import because not all PyTorch builds have torchgen
+    from torchgen.model import FunctionSchema, SchemaKind
```

Comment:
Would it make more sense to use https://github.com/pytorch/pytorch/blob/0ea9646cdddce2339dfea998340c55323d617337/torch/csrc/jit/python/init.cpp#L1717 here ?